### ISSUE TITLE
WordDumb can't find Python (installed via Homebrew), even after specifying explicitly the path to it via macos-env.txt

### ISSUE THREAD

**Issue by arooni** [xxyzz/WordDumb#137]
```Starting job: Generating Word Wise and X-Ray for Meditations 
Job: "Generating Word Wise and X-Ray for Meditations" failed with error: 
Traceback (most recent call last):
  File "calibre/gui2/threaded_jobs.py", line 82, in start_work
  File "calibre_plugins.worddumb.parse_job", line 154, in do_job
    install_deps("lxml", notifications)
  File "calibre_plugins.worddumb.deps", line 36, in install_deps
    PY_PATH, py_version = which_python()
  File "calibre_plugins.worddumb.deps", line 84, in which_python
    raise Exception("PythonNotFound")
Exception: PythonNotFound
 
Called with args: ((411, 'EPUB', '/Users/david/syncthing/ebooks/calibre library/Marcus Aurelius (Emperor of Rome)/Meditations (411)/Meditations - Marcus Aurelius (Emperor of Rome).epub', <calibre.ebooks.metadata.book.base.Metadata object at 0x116ee72e0>, 'en'), True, True) {'notifications': <queue.Queue object at 0x116ee6440>, 'abort': <threading.Event object at 0x116ee7df0>, 'log': <calibre.utils.logging.GUILog object at 0x116ee60b0>} 
```

Despite this being in [~/Library/Preferences/calibre/macos-env.txt:

```
PATH=/opt/homebrew/opt/python@3.11/libexec/bin:$PATH

```
```
[I]  /   master   which python
/usr/local/opt/python@3.11/libexec/bin/python
```

**Comment by xxyzz** [comment]
You don't have to add PATH to macos-env.txt if the package is installed from homebrew, and please also make sure both python and calibre are not using Rosetta.

**Comment by arooni** [comment]
Thanks for the fast response time !

M1 mac would mean no Rosetta correct ?

And I tried to add it to that file but the only way I could make your
plugin work was launch it from the command line in fish where the path is
set to see python.

**Comment by arooni** [comment]
I did install python from homebrew.

**Comment by xxyzz** [comment]
Please read this document(https://support.apple.com/en-us/HT211861) to check if calibre is using Rosetta.

And homebrew will link python to the correct location(/usr/local/bin or /opt/homebrew/bin) so users shouldn't add the path manually. Could you check if `/opt/homebrew/bin/python3` exists?

And also the macos-env.txt file feature is not included in the v3.29.3 release. You can download the master branch test version from GitHub Actions Artifacts.
